edx_rest_framework_extensions/auth/jwt/middleware.py

```diff
@@ -40,29 +41,45 @@ class EnsureJWTAuthSettingsMiddleware(MiddlewareMixin):
     """
     _required_permission_classes = (NotJwtRestrictedApplication,)
 
+    @staticmethod
+    def _get_decomposed_permissions(permission):
+        """
+        Given a composed permission class or instance, return its component parts.
+
+        Examples:
+
+        Composition using DRF native support in 3.9+:
+            IsStaff | IsSuperuser -> [IsStaff, IsSuperuser]
+            IsOwner | IsStaff | IsSuperuser -> [IsOwner | IsStaff, IsSuperuser]
+
+        """
+        # We skip SingleOperandHolder since it contains negations
+        if isinstance(permission, OperandHolder):
+            return [permission.op1_class, permission.op2_class]
+        return []
+
+    def _iter_included_base_classes(self, view_permissions):
+        # Not all permissions are classes, some will be ConditionalPermission
+        # objects from the rest_condition library, and others will be OperandHolder
```

Comment:
Not sure if this comment about `rest_condition` is current or not, but I have some related questions.
1. Composed conditions may have been created outside of edx-drf-extensions that will pass through this middleware, right? Will this middleware need to complain about that? Will it work?
2. Will `rest_condition` still need to be a test dependency for a while to ensure we do whatever we decide we should be doing in this case?
3. Based on our decisions, is this a backward incompatible change that requires a major version bump?

Reply:
It is not current. For a while I kept compatibility with both, and included it as a test dependency, however I realised that since rest condition isn't compatible with django 3+ and I was removing the usage from the platform, keeping the test around didn't make much sense. 

It is a backwards incompatible change, since without handling rest condition, the platform with just crash if it uses rest condition or uses any plugins that use it.